src/python/pants/backend/python/rules/python_test_runner_integration_test.py

```diff
@@ -1,266 +1,376 @@
+    )
+    # TODO: replace all this boilerplate with a utility to set up OptionsBootstrapper for
+    # non-ConsoleRuleTestBase tests.
+    mock_global_options = OptionValueContainer()
+    mock_global_options.colors = RankedValue(RankedValue.HARDCODED, False)
+    test_result = self.request_single_product(
+      TestResult,
+      Params(
+        target,
+        test_target_setup,
+        PythonSetup.global_instance(),
+        SubprocessEnvironment.global_instance(),
+        GlobalOptions(mock_global_options),
       )
-
-  def test_source_dep_relative_import(self):
-    pants_run = self.run_passing_pants_test([
-      'testprojects/tests/python/pants/dummies:target_with_source_dep_relative_import',
-    ])
-    self.assert_fuzzy_string_match(
-      pants_run.stdout_data,
-      dedent("""\
-        testprojects/tests/python/pants/dummies:target_with_source_dep_relative_import stdout:
-        ============================= test session starts ==============================
-        platform SOME_TEXT
-        rootdir: SOME_TEXT
-        plugins: SOME_TEXT
-        collected 1 item
-
-        pants/dummies/test_with_source_dep_relative_import.py .                  [100%]
-
-        ============================== 1 passed in SOME_TEXT ===============================
-
-
-        testprojects/tests/python/pants/dummies:target_with_source_dep_relative_import  .....   SUCCESS
-        """)
+    )
+    debug_result = self.request_single_product(
+      TestDebugResult, Params(test_target_setup, InteractiveRunner(self.scheduler)),
+    )
+    if test_result.status == Status.SUCCESS:
+      assert debug_result.exit_code == 0
+    else:
+      assert debug_result.exit_code != 0
+    return test_result
+
+  def test_single_passing_test(self) -> None:
+    self.create_python_test_target([self.good_source])
+    result = self.run_pytest()
+    assert result.status == Status.SUCCESS
+    assert "test_good.py ." in result.stdout
+
+  def test_single_failing_test(self) -> None:
+    self.create_python_test_target([self.bad_source])
+    result = self.run_pytest()
+    assert result.status == Status.FAILURE
+    assert "test_bad.py F" in result.stdout
+
+  def test_mixed_sources(self) -> None:
+    self.create_python_test_target([self.good_source, self.bad_source])
+    result = self.run_pytest()
+    assert result.status == Status.FAILURE
+    assert "test_good.py ." in result.stdout
+    assert "test_bad.py F" in result.stdout
+
+  def test_absolute_import(self) -> None:
+    self.create_basic_library()
+    source = FileContent(
+      path="test_absolute_import.py",
+      content=dedent(
+        """\
+        from pants_test.library import add_two
+
+        def test():
+          assert add_two(2) == 4
+        """
+      ).encode(),
+    )
+    self.create_python_test_target([source], dependencies=[":library"])
+    result = self.run_pytest()
+    assert result.status == Status.SUCCESS
+    assert "test_absolute_import.py ." in result.stdout
```

Comment:
This test is repetitive. It's a simpler case of both of these:

* `test_transitive_dep`
* `test_thirdparty_transitive_dep`

Should we delete this? Even once we fix the Pex performance issue, these tests will still not be free.